droidlet/perception/robot/perception.py

```diff
@@ -98,14 +95,16 @@ def perceive(self, rgb_depth, xyz, previous_objects, force=False):
             old_image, detections, humans, old_xyz = None, None, None, None
 
         self.log(rgb_depth, detections, humans, old_image)
+        perception_output = {}
         if detections is not None:
-            current_objects = detections + humans
             if previous_objects is not None:
-                new_objects, updated_objects = self.vision.deduplicate(
-                    current_objects, previous_objects
+                new_detections, updated_detections = self.vision.deduplicate(
+                    detections, previous_objects
                 )
-            return (new_objects, updated_objects)
-        return None
+                perception_output["new_detections"] = new_detections
+                perception_output["updated_detections"] = updated_detections
+        perception_output["humans"] = humans
+        return perception_output
```

Comment:
dont return a complex dict, just return the three outputs separately

Reply:
Fixed in latest commit